src/data_device/mod.rs

```diff
@@ -279,8 +279,5 @@ where
 }
 
 pub(crate) fn no_nix_err(err: nix::Error) -> std::io::Error {
-    match err {
-        nix::Error::Sys(e) => e.into(),
-        _ => unreachable!("Impossible error: {:?}", err),
-    }
+    err.into()
 }
```

Comment:
If there is now an Into<std::io::Error> impl for nix::Errno, then this no_nix_err function should just be removed, as it no longer serves any purpose.